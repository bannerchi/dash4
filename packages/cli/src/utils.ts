import fs from 'fs-extra';
import path from 'path';

export interface IPackageJson {
	name?: string;
	version?: string;
	description?: string;
	dependencies?: { [key: string]: string };
	devDependencies?: { [key: string]: string };
	[key: string]: any;
}
export const getScript = (packageData: IPackageJson, scriptName: string) => {
	if (!hasScript(packageData, scriptName)) {
		return;
	}
	return packageData.scripts[scriptName];
};
export const hasScript = (packageData: IPackageJson, scriptName: string) =>
	packageData.scripts && packageData.scripts[scriptName] && packageData.scripts[scriptName] !== '';
export const hasDependency = (packageData: IPackageJson, dependencyName: string) =>
	(packageData.devDependencies && packageData.devDependencies[dependencyName]) ||
	(packageData.dependencies && packageData.dependencies[dependencyName]);

export async function hasFile(...pathFragments: string[]) {
	try {
		return (await fs.stat(path.join(...pathFragments))).isFile();
	} catch (err) {
		return false;
	}
}
