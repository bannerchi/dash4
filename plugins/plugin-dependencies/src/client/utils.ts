import { IDependencyObj, IError } from '../shared-types';

export function isDataLoading(value: IDependencyObj | IError | undefined): value is undefined {
	return typeof value === 'undefined';
}

export function isDataError(value: IDependencyObj | IError): value is IError {
	return typeof value === 'object' && Object.keys(value).includes('error');
}
