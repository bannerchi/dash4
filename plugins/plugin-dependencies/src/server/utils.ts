export async function asyncForEach<TArrItem = any>(array: TArrItem[], onEach: (item: TArrItem, index: number) => void) {
	for (let index = 0; index < array.length; index++) {
		// eslint-disable-next-line
		await onEach(array[index], index);
	}
}
