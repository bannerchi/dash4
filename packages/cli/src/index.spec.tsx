/// <reference types="@types/jest" />
// eslint-disable-next-line
import React from 'react';
import renderer from 'react-test-renderer';

const execaMock = jest.fn();
jest.mock('execa', () => ({
	__esModule: true,
	default: execaMock,
}));
const spinnerFailMock = jest.fn();
jest.mock('@dash4/log', () => ({
	__esModule: true,
	spinner: () => {
		return {
			start: jest.fn(),
			succeed: jest.fn(),
			fail: spinnerFailMock,
			text: jest.fn(),
		};
	},
}));
jest.mock('chalk', () => ({
	__esModule: true,
	default: {
		// eslint-disable-next-line
		hex: (vaui: string) => (text: string) => text,
		bgBlack: (text: string) => text,
		white: (text: string) => text,
		red: (text: string) => text,
		bold: (text: string) => text,
	},
}));

import cpy from 'cpy';
import del from 'del';
import fs from 'fs-extra';
import makeDir from 'make-dir';
import path from 'path';
import readPkg from 'read-pkg';
import { init } from '.';
import { hasFile } from './utils';

const tempDir = path.join(__dirname, '../__tmp__');
const getTempFile = async (testId: string, fileName: string) =>
	await fs.readFile(path.join(tempDir, testId, fileName), 'utf8');
const mockDir = path.join(__dirname, '../src/__mocks__');
const isCI = ((process.env || {}).NODE_ENV || '').toLowerCase() === 'ci';

const processKill = process.kill;
// eslint-disable-next-line
const consoleLog = console.log;

process.kill = jest.fn();
// eslint-disable-next-line
console.log = jest.fn();

const getOptions = (cwd: string, force = true) => ({
	port: 4000,
	config: path.join(cwd, 'dash4.config.js'),
	force,
});

async function runSnapshotTest(testId: string, dependencies: string) {
	const cwd = path.posix.join(tempDir, testId);
	await makeDir(cwd);
	await cpy([`src/__mocks__/${testId}/_package.json`], `__tmp__/${testId}`, {
		rename: () => `package.json`,
	});
	if (await hasFile(mockDir, testId, `README.md`)) {
		await cpy([`src/__mocks__/${testId}/README.md`], `__tmp__/${testId}`);
	}
	if (await hasFile(mockDir, testId, `readme.md`)) {
		await cpy([`src/__mocks__/${testId}/readme.md`], `__tmp__/${testId}`);
	}
	await init(cwd, getOptions(cwd));
	expect(execaMock).toHaveBeenCalledWith('npm', `i -D @dash4/server ${dependencies}`.split(' '));
	expect((((await readPkg({ cwd })) || {}) as any).scripts.dash4).toBe('dash4');
	const fileData = await getTempFile(testId, 'dash4.config.js');
	const tree = renderer.create(<div dangerouslySetInnerHTML={{ __html: fileData }} />).toJSON();
	expect(tree).toMatchSnapshot();
}

describe('cli', () => {
	beforeEach(() => {
		jest.resetAllMocks();
	});

	beforeAll(async () => {
		await del(tempDir);
		// create needed directories and files
		await makeDir(tempDir);
	});

	afterAll(async () => {
		// eslint-disable-next-line
		console.log = consoleLog;
		process.kill = processKill;
		// await del(tempDir);
	});

	test('is function', async () => {
		expect(typeof init).toBe('function');
	});
	test('execute in empty directory should fail with error', async () => {
		const cwd = path.join(tempDir, 'empty');
		await makeDir(cwd);

		const spyProcess = jest.spyOn(process, 'kill');
		await init(cwd, getOptions(cwd));
		expect(spinnerFailMock).toHaveBeenCalledWith('package.json not found!');
		expect(spyProcess).toHaveBeenCalledWith(1);
	});
	test('execute in directory where config exist', async () => {
		const cwd = path.join(tempDir, 'config-exist');
		await makeDir(cwd);
		await fs.writeFile(path.join(cwd, 'package.json'), '{ "scripts": { "start": "dash4" } }');
		await fs.writeFile(path.join(cwd, 'dash4.config.js'), '');

		const spyProcess = jest.spyOn(process, 'kill');
		await init(cwd, getOptions(cwd, false));
		expect(spinnerFailMock).toHaveBeenCalledWith('Dash4 is already installed!');
		expect(spyProcess).toHaveBeenCalledWith(1);
	});
	test('execute in directory where config exist but with force mode', async () => {
		const cwd = path.join(tempDir, 'config-exist');
		await makeDir(cwd);
		await fs.writeFile(path.join(cwd, 'package.json'), '{ "scripts": { "start": "dash4" } }');
		await fs.writeFile(path.join(cwd, 'dash4.config.js'), '');
		await init(cwd, getOptions(cwd, true));
		expect((((await readPkg({ cwd })) || {}) as any).scripts.dash4).toBe('dash4');
	});
	test('execute with script "start"', async () => {
		await runSnapshotTest('script-start', '@dash4/plugin-dependencies @dash4/plugin-terminal');
	});
	test('execute with "README.md" file', async () => {
		expect(await hasFile(mockDir, 'readme', `README.md`)).toBe(true);
		await runSnapshotTest('readme', '@dash4/plugin-dependencies @dash4/plugin-readme');
	});
	test('execute with script "test"', async () => {
		await runSnapshotTest(
			'script-test',
			'@dash4/plugin-dependencies @dash4/plugin-code-coverage @dash4/plugin-npm-scripts'
		);
	});
	test('execute with script "watch-test"', async () => {
		await runSnapshotTest(
			'script-watch-test',
			'@dash4/plugin-dependencies @dash4/plugin-terminal @dash4/plugin-code-coverage'
		);
	});
	test('execute with script "watch-test-jest"', async () => {
		await runSnapshotTest(
			'script-watch-test-jest',
			'@dash4/plugin-dependencies @dash4/plugin-terminal @dash4/plugin-code-coverage'
		);
	});
	test('execute in create-react-app setup', async () => {
		await runSnapshotTest(
			'script-cra',
			'@dash4/plugin-dependencies @dash4/plugin-terminal @dash4/plugin-code-coverage @dash4/plugin-npm-scripts'
		);
	});
	test('execute with script "storybook"', async () => {
		await runSnapshotTest('script-storybook', '@dash4/plugin-dependencies @dash4/plugin-terminal');
	});
	test('execute with all possible scripts', async () => {
		await runSnapshotTest(
			'all-scripts',
			'@dash4/plugin-dependencies @dash4/plugin-terminal @dash4/plugin-readme @dash4/plugin-code-coverage @dash4/plugin-npm-scripts'
		);
	});

	(isCI ? test.skip : test)('execute in lerna repo', async () => {
		const testId = 'lerna';
		const cwd = path.join(tempDir, testId);
		await makeDir(cwd);

		// add empty dir
		await makeDir(path.join(cwd, 'packages/empty'));
		await cpy([`src/__mocks__/${testId}/_package.json`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
			rename: () => `package.json`,
		});
		await cpy([`src/__mocks__/${testId}/README.md`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
		});
		await cpy([`src/__mocks__/${testId}/_lerna.json`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
			rename: () => `lerna.json`,
		});

		// create configs directory
		await makeDir(path.join(cwd, 'configs'));
		try {
			await cpy([`*/_package.json`], `../../../../__tmp__/${testId}/configs/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/configs`),
				parents: true,
				rename: () => `package.json`,
			});
		} catch (err) {
			// eslint-disable-next-line
		}
		try {
			await cpy([`*/README.md`], `../../../../__tmp__/${testId}/configs/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/configs`),
				parents: true,
			});
		} catch (err) {
			// eslint-disable-next-line
		}

		// create packages directory
		await makeDir(path.join(cwd, 'packages'));
		try {
			await cpy([`*/_package.json`], `../../../../__tmp__/${testId}/packages/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/packages`),
				parents: true,
				rename: () => `package.json`,
			});
		} catch (err) {
			// eslint-disable-next-line
		}
		try {
			await cpy([`*/README.md`], `../../../../__tmp__/${testId}/packages/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/packages`),
				parents: true,
			});
		} catch (err) {
			// eslint-disable-next-line
		}

		await init(cwd, getOptions(cwd));

		const fileData = await getTempFile(testId, 'dash4.config.js');
		const tree = renderer.create(<div dangerouslySetInnerHTML={{ __html: fileData }} />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(execaMock).toHaveBeenCalledWith(
			'npm',
			`i -D @dash4/server @dash4/plugin-dependencies @dash4/plugin-readme @dash4/plugin-code-coverage @dash4/plugin-npm-scripts @dash4/plugin-terminal`.split(
				' '
			)
		);
	});

	test.skip('execute in yarn workspaces repo', async () => {
		const testId = 'yarn-workspaces';
		const cwd = path.join(tempDir, testId);
		await makeDir(cwd);

		// add empty dir
		await makeDir(path.join(cwd, 'packages/empty'));
		await cpy([`src/__mocks__/${testId}/_package.json`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
			rename: () => `package.json`,
		});
		await cpy([`src/__mocks__/${testId}/README.md`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
		});
		await cpy([`src/__mocks__/${testId}/_lerna.json`], `__tmp__/${testId}`, {
			cwd: path.join(__dirname, '..'),
			rename: () => `lerna.json`,
		});

		// create configs directory
		await makeDir(path.join(cwd, 'configs'));
		try {
			await cpy([`*/_package.json`], `../../../../__tmp__/${testId}/configs/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/configs`),
				parents: true,
				rename: () => `package.json`,
			});
		} catch (err) {
			// eslint-disable-next-line
		}
		try {
			await cpy([`*/README.md`], `../../../../__tmp__/${testId}/configs/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/configs`),
				parents: true,
			});
		} catch (err) {
			// eslint-disable-next-line
		}

		// create packages directory
		await makeDir(path.join(cwd, 'packages'));
		try {
			await cpy([`*/_package.json`], `../../../../__tmp__/${testId}/packages/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/packages`),
				parents: true,
				rename: () => `package.json`,
			});
		} catch (err) {
			// eslint-disable-next-line
		}
		try {
			await cpy([`*/README.md`], `../../../../__tmp__/${testId}/packages/`, {
				cwd: path.join(__dirname, `__mocks__/${testId}/packages`),
				parents: true,
			});
		} catch (err) {
			// eslint-disable-next-line
		}

		await init(cwd, getOptions(cwd));

		const fileData = await getTempFile(testId, 'dash4.config.js');
		const tree = renderer.create(<div dangerouslySetInnerHTML={{ __html: fileData }} />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(execaMock).toHaveBeenCalledWith(
			'yarn',
			`add -D -W @dash4/server @dash4/plugin-dependencies @dash4/plugin-terminal @dash4/plugin-readme @dash4/plugin-code-coverage @dash4/plugin-npm-scripts`.split(
				' '
			)
		);
	});
});
