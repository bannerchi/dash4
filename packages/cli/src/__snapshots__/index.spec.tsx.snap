// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli execute in create-react-app setup 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-npm-scripts
const { PluginNpmScripts } = require('@dash4/plugin-npm-scripts');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[new PluginDependencies(), new PluginTerminal({ cmd: 'npm start', autostart: true })],
					[
						new PluginTerminal({ cmd: 'npm run test', allowedCommands: jestCommands }),
						new PluginCodeCoverage(),
					],
					[
						new PluginNpmScripts({
							scripts: [{ title: 'build', cmd: 'npm run build' }, { title: 'test', cmd: 'npm run test' }],
						}),
					],
				],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute in lerna repo 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-readme
const { PluginReadme } = require('@dash4/plugin-readme');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-npm-scripts
const { PluginNpmScripts } = require('@dash4/plugin-npm-scripts');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[
						new PluginDependencies({
							lerna: 'lerna.json',
							installProcess: { title: 'run bootstrap', cmd: 'npm run bootstrap', cwd: '/' },
						}),
						new PluginReadme({ file: 'README.md' }),
					],
					[
						new PluginCodeCoverage(),
						new PluginNpmScripts({
							scripts: [
								{ title: 'bootstrap', cmd: 'npm run bootstrap' },
								{ title: 'build', cmd: 'npm run build' },
								{ title: 'test', cmd: 'npm run test' },
								{ title: 'lint', cmd: 'npm run lint' },
							],
						}),
					],
				],
			},
			{
				title: 'script-start',
				rows: [[new PluginTerminal({ cmd: 'npm start', cwd: 'packages/script-start', autostart: true })]],
			},
			{
				title: 'script-storybook',
				rows: [[new PluginTerminal({ cmd: 'npm run storybook', cwd: 'packages/script-storybook' })]],
			},
			{
				title: 'script-test',
				rows: [
					[
						new PluginCodeCoverage({ cwd: 'packages/script-test' }),
						new PluginNpmScripts({
							scripts: [{ title: 'test', cmd: 'npm run test', cwd: 'packages/script-test' }],
						}),
					],
				],
			},
			{
				title: 'script-watch-test',
				rows: [
					[
						new PluginTerminal({ cmd: 'npm run watch-test', cwd: 'packages/script-watch-test' }),
						new PluginCodeCoverage({ cwd: 'packages/script-watch-test' }),
					],
				],
			},
			{
				title: 'all-scripts',
				rows: [
					[
						new PluginTerminal({ cmd: 'npm start', cwd: 'configs/all-scripts', autostart: true }),
						new PluginTerminal({ cmd: 'npm run storybook', cwd: 'configs/all-scripts' }),
					],
					[
						new PluginReadme({ file: 'configs/all-scripts/README.md' }),
						new PluginTerminal({ cmd: 'npm run watch-test', cwd: 'configs/all-scripts' }),
					],
					[
						new PluginCodeCoverage({ cwd: 'configs/all-scripts' }),
						new PluginNpmScripts({
							scripts: [
								{ title: 'bootstrap', cmd: 'npm run bootstrap', cwd: 'configs/all-scripts' },
								{ title: 'build', cmd: 'npm run build', cwd: 'configs/all-scripts' },
								{ title: 'watch', cmd: 'npm run watch', cwd: 'configs/all-scripts' },
								{ title: 'test', cmd: 'npm run test', cwd: 'configs/all-scripts' },
								{ title: 'lint', cmd: 'npm run lint', cwd: 'configs/all-scripts' },
								{ title: 'clean', cmd: 'npm run clean', cwd: 'configs/all-scripts' },
								{ title: 'flow', cmd: 'npm run flow', cwd: 'configs/all-scripts' },
								{ title: 'compile', cmd: 'npm run compile', cwd: 'configs/all-scripts' },
								{ title: 'format', cmd: 'npm run format', cwd: 'configs/all-scripts' },
								{ title: 'prettier', cmd: 'npm run prettier', cwd: 'configs/all-scripts' },
								{ title: 'coverage', cmd: 'npm run coverage', cwd: 'configs/all-scripts' },
								{ title: 'prepare', cmd: 'npm run prepare', cwd: 'configs/all-scripts' },
								{ title: 'validate', cmd: 'npm run validate', cwd: 'configs/all-scripts' },
								{ title: 'deploy', cmd: 'npm run deploy', cwd: 'configs/all-scripts' },
								{ title: 'docs', cmd: 'npm run docs', cwd: 'configs/all-scripts' },
								{
									title: 'build-storybook',
									cmd: 'npm run build-storybook',
									cwd: 'configs/all-scripts',
								},
							],
						}),
					],
				],
			},
			{
				title: 'readme',
				rows: [[new PluginReadme({ file: 'configs/readme/README.md' })]],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute in yarn workspaces repo 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-readme
const { PluginReadme } = require('@dash4/plugin-readme');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-npm-scripts
const { PluginNpmScripts } = require('@dash4/plugin-npm-scripts');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[
						new PluginDependencies({
							lerna: 'lerna.json',
							installProcess: { title: 'run bootstrap', cmd: 'yarn bootstrap', cwd: '/' },
						}),
						new PluginTerminal({ cmd: 'yarn start', autostart: true }),
					],
					[new PluginReadme({ file: 'README.md' }), new PluginCodeCoverage()],
					[
						new PluginNpmScripts({
							scripts: [
								{ title: 'bootstrap', cmd: 'yarn bootstrap' },
								{ title: 'build', cmd: 'yarn build' },
								{ title: 'test', cmd: 'yarn test' },
								{ title: 'lint', cmd: 'yarn lint' },
							],
						}),
					],
				],
			},
			{
				title: 'script-start',
				rows: [[new PluginTerminal({ cmd: 'yarn start', cwd: 'packages/script-start', autostart: true })]],
			},
			{
				title: 'script-storybook',
				rows: [[new PluginTerminal({ cmd: 'yarn storybook', cwd: 'packages/script-storybook' })]],
			},
			{
				title: 'script-test',
				rows: [
					[
						new PluginCodeCoverage({ cwd: 'packages/script-test' }),
						new PluginNpmScripts({
							scripts: [{ title: 'test', cmd: 'yarn test', cwd: 'packages/script-test' }],
						}),
					],
				],
			},
			{
				title: 'script-watch-test',
				rows: [
					[
						new PluginTerminal({ cmd: 'yarn watch-test', cwd: 'packages/script-watch-test' }),
						new PluginCodeCoverage({ cwd: 'packages/script-watch-test' }),
					],
				],
			},
			{
				title: 'all-scripts',
				rows: [
					[
						new PluginTerminal({ cmd: 'yarn start', cwd: 'configs/all-scripts', autostart: true }),
						new PluginTerminal({ cmd: 'yarn storybook', cwd: 'configs/all-scripts' }),
					],
					[
						new PluginReadme({ file: 'configs/all-scripts/README.md' }),
						new PluginTerminal({ cmd: 'yarn watch-test', cwd: 'configs/all-scripts' }),
					],
					[
						new PluginCodeCoverage({ cwd: 'configs/all-scripts' }),
						new PluginNpmScripts({
							scripts: [
								{ title: 'bootstrap', cmd: 'yarn bootstrap', cwd: 'configs/all-scripts' },
								{ title: 'build', cmd: 'yarn build', cwd: 'configs/all-scripts' },
								{ title: 'watch', cmd: 'yarn watch', cwd: 'configs/all-scripts' },
								{ title: 'test', cmd: 'yarn test', cwd: 'configs/all-scripts' },
								{ title: 'lint', cmd: 'yarn lint', cwd: 'configs/all-scripts' },
								{ title: 'clean', cmd: 'yarn clean', cwd: 'configs/all-scripts' },
								{ title: 'flow', cmd: 'yarn flow', cwd: 'configs/all-scripts' },
								{ title: 'compile', cmd: 'yarn compile', cwd: 'configs/all-scripts' },
								{ title: 'format', cmd: 'yarn format', cwd: 'configs/all-scripts' },
								{ title: 'prettier', cmd: 'yarn prettier', cwd: 'configs/all-scripts' },
								{ title: 'coverage', cmd: 'yarn coverage', cwd: 'configs/all-scripts' },
								{ title: 'prepare', cmd: 'yarn prepare', cwd: 'configs/all-scripts' },
								{ title: 'validate', cmd: 'yarn validate', cwd: 'configs/all-scripts' },
								{ title: 'deploy', cmd: 'yarn deploy', cwd: 'configs/all-scripts' },
								{ title: 'docs', cmd: 'yarn docs', cwd: 'configs/all-scripts' },
								{ title: 'build-storybook', cmd: 'yarn build-storybook', cwd: 'configs/all-scripts' },
							],
						}),
					],
				],
			},
			{
				title: 'readme',
				rows: [[new PluginReadme({ file: 'configs/readme/README.md' })]],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with "README.md" file 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-readme
const { PluginReadme } = require('@dash4/plugin-readme');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [[new PluginDependencies(), new PluginReadme({ file: 'README.md' })]],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with all possible scripts 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-readme
const { PluginReadme } = require('@dash4/plugin-readme');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-npm-scripts
const { PluginNpmScripts } = require('@dash4/plugin-npm-scripts');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[new PluginDependencies(), new PluginTerminal({ cmd: 'npm start', autostart: true })],
					[new PluginTerminal({ cmd: 'npm run storybook' }), new PluginReadme({ file: 'README.md' })],
					[new PluginTerminal({ cmd: 'npm run watch-test' }), new PluginCodeCoverage()],
					[
						new PluginNpmScripts({
							scripts: [
								{ title: 'bootstrap', cmd: 'npm run bootstrap' },
								{ title: 'build', cmd: 'npm run build' },
								{ title: 'watch', cmd: 'npm run watch' },
								{ title: 'test', cmd: 'npm run test' },
								{ title: 'lint', cmd: 'npm run lint' },
								{ title: 'clean', cmd: 'npm run clean' },
								{ title: 'flow', cmd: 'npm run flow' },
								{ title: 'compile', cmd: 'npm run compile' },
								{ title: 'format', cmd: 'npm run format' },
								{ title: 'prettier', cmd: 'npm run prettier' },
								{ title: 'coverage', cmd: 'npm run coverage' },
								{ title: 'prepare', cmd: 'npm run prepare' },
								{ title: 'validate', cmd: 'npm run validate' },
								{ title: 'deploy', cmd: 'npm run deploy' },
								{ title: 'docs', cmd: 'npm run docs' },
								{ title: 'build-storybook', cmd: 'npm run build-storybook' },
							],
						}),
					],
				],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with script "start" 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [[new PluginDependencies(), new PluginTerminal({ cmd: 'npm start', autostart: true })]],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with script "storybook" 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [[new PluginDependencies(), new PluginTerminal({ cmd: 'npm run storybook' })]],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with script "test" 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-npm-scripts
const { PluginNpmScripts } = require('@dash4/plugin-npm-scripts');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[new PluginDependencies(), new PluginCodeCoverage()],
					[new PluginNpmScripts({ scripts: [{ title: 'test', cmd: 'npm run test' }] })],
				],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with script "watch-test" 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[new PluginDependencies(), new PluginTerminal({ cmd: 'npm run watch-test' })],
					[new PluginCodeCoverage()],
				],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;

exports[`cli execute with script "watch-test-jest" 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "/**
 *
 * 	DASH4 configuration
 *  https://github.com/smollweide/dash4
 *
 */
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-dependencies
const { PluginDependencies } = require('@dash4/plugin-dependencies');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-terminal
const { PluginTerminal, jestCommands } = require('@dash4/plugin-terminal');
// https://github.com/smollweide/dash4/tree/master/plugins/plugin-code-coverage
const { PluginCodeCoverage } = require('@dash4/plugin-code-coverage');

async function getConfig() {
	return {
		port: 8080,
		tabs: [
			{
				title: 'Root',
				rows: [
					[
						new PluginDependencies(),
						new PluginTerminal({ cmd: 'npm run watch-test', allowedCommands: jestCommands }),
					],
					[new PluginCodeCoverage()],
				],
			},
		],
	};
}

module.exports = getConfig;
",
    }
  }
/>
`;
